### PR TITLE
WIP: Display Map Depending on the Forecast

### PR DESCRIPTION
**Pull request by Jargendas**
Hi Jalibu,
great module, I love it! I also like the "displayOnlyOnRain" option, however, I noticed that the map is usually shown too late with the option enabled for my liking. I especially need the map when it does not rain just yet, to estimate if it makes sense to quickly go outside before the rain or stay inside until it passed by.

As I have an hourly forecast (default weather module with OpenWeatherMap) running anyways, I implemented a feature based on the emitted 'WEATHER_UPDATED' notification and replaced the boolean "displayOnlyOnRain" option with a numeric "displayHoursBeforeRain". As the name suggests, the module will check if it will (by forecast) rain within the given number of hours, and if it will, display the module - or otherwise hide it. Setting the value to Infinity or 0 will have the same effect as the "displayOnlyOnRain" option before, so no functionality is lost. 

I just implemented it today and still have to do some more tests to make sure everything works as expected, but I already wanted to ask what you think about this idea and if you are willing to integrate it. :)

### CODE REVIEW

**Comment by jalibu** [comment]
Me gusta!

It brings great value when the map announces the rain BEFORE it is there :-D

Thanks for your contribution. I'll take a look at it over the next week.

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ https://user-images.githubusercontent.com/25933231/130909536-e096d342-19d5-4139-
    		displayTime: true,
    		displayTimeline: true,
    		displayClockSymbol: true,
-   		displayOnlyOnRain: false,
+   		displayHoursBeforeRain: Infinity,
```
Comment (jalibu) [inline README.md:53]: Even if Infinity is syntactically correct, could we use -1 for "show always" instead?

File: src/frontend/Frontend.ts [hunk 2]
```diff
@@ -298,19 +298,47 @@ Module.register<Config>('MMM-RAIN-MAP', {
...
+      } else if (this.config.displayHoursBeforeRain > 0) {
+        if (notificationIdentifier === 'WEATHER_UPDATED') {
+          this.handleWeatherUpdate(payload)
+        }
+      }
     }
   },
 
+  handleWeatherUpdate(update: any) {
+    Log.log('Weather update!')
```
Comment (jalibu) [inline src/frontend/Frontend.ts:322]: please remove

File: src/frontend/Frontend.ts [hunk 3]
```diff
@@ -298,19 +298,47 @@ Module.register<Config>('MMM-RAIN-MAP', {
...
+    const hourlyData = update.hourlyArray
+    var closestRain = Infinity
```
Comment (jalibu) [inline src/frontend/Frontend.ts:324]: Please use "let" instead of "var"

File: src/frontend/Frontend.ts [hunk 4]
```diff
@@ -298,19 +298,47 @@ Module.register<Config>('MMM-RAIN-MAP', {
...
+    const now = Date.now()
+    for (let entry of hourlyData) {
```
Comment (jalibu) [inline src/frontend/Frontend.ts:326]: Please use const, not let

**Comment by jalibu** [comment]
@Jargendas can you please open a PR against dev?

File: README.md (hunk 1 above)
Comment (Jargendas) [reply README.md:53]: Yes, good idea

**Comment by Jargendas** [comment]
Okay, I integrated your comments and changed the PR to be against dev.
Last week, it rained a lot where I live and the map appeared always when I needed it, so it seems to work well. :)

**Pull request merged**